### PR TITLE
Converted CreateQueue to Functional Component

### PR DESCRIPTION
**Pull request by PavanDongare**
unsure of placement of : store.dispatch(setCreationStep(0))

### CODE REVIEW

File: simplq/src/components/pages/CreateQueue.jsx [hunk 1]
```diff
@@ -11,76 +11,66 @@ import { SimplQHeader } from '../common/Header';
...
+      <InputField
+        placeholder="Enter a name for your new queue"
+        value={textFieldValue}
+        onChange={handleTextFieldChange}
+        onKeyPress={
+          (e) => handleEnterPress(e, () => handleClick(textFieldValue))
+          // eslint-disable-next-line react/jsx-curly-newline
+        }
+        error={invalidMsg.length > 0}
+        helperText={invalidMsg.length > 0 ? invalidMsg : ''}
```
Comment (daltonfury42) [inline simplq/src/components/pages/CreateQueue.jsx:62]: Just wondering, we can just give `invalidMsg` here, will that work?

**Comment by daltonfury42** [comment]
> unsure of placement of : store.dispatch(setCreationStep(0))

this is fine... Lets meet sometime tomorrow. I can show you around the code a bit, and give you some overview.

File: simplq/src/components/pages/CreateQueue.jsx [hunk 2]
```diff
@@ -11,76 +11,66 @@ import { SimplQHeader } from '../common/Header';
...
-      this.setState({ createInProgress: true });
+      setCreateInProgress(true);
       QueueService.create(queueName)
         .then((response) => {
           store.dispatch(setQueueId(response.queueId));
           store.dispatch(setQueueName(response.queueName));
           store.dispatch(setCreationStep(1));
-          this.props.history.push('/admin');
+          history.push('/admin');
         })
```
Comment (maaverik) [inline simplq/src/components/pages/CreateQueue.jsx:29]: This would conflict with https://github.com/SimplQ/simplQ-frontend/pull/160

Comment (maaverik) [reply simplq/src/components/pages/CreateQueue.jsx:29]: Could you change it to history.push(`/queue/{response.queueId}`) and remove the store.dispatch(setQueueId(response.queueId));

Comment (maaverik) [reply simplq/src/components/pages/CreateQueue.jsx:29]: Best to wait till #160 is merged before this change though, to atleast test and make sure we're not missing aything else

File: simplq/src/components/pages/CreateQueue.jsx [hunk 3]
```diff
@@ -11,76 +11,66 @@ import { SimplQHeader } from '../common/Header';
...
-    if (isQueueNameValid(qname))
-      this.setState({
-        textFieldValue: qname,
-        invalidMsg: '',
-      });
-    else {
-      this.setState({
-        invalidMsg: "Only alphabets, numbers and '-' allowed",
-      });
+    if (isQueueNameValid(qname)) {
```
[the remark this answers is not in the document]
Comment (PavanDongare) [reply simplq/src/components/pages/CreateQueue.jsx:39]: changed

File: simplq/src/components/pages/CreateQueue.jsx (hunk 1 above)
Comment (PavanDongare) [reply simplq/src/components/pages/CreateQueue.jsx:62]: not sure if any scenario when invalid msg is NOT a string. So should work.

File: simplq/src/components/pages/CreateQueue.jsx (hunk 2 above)
Comment (PavanDongare) [reply simplq/src/components/pages/CreateQueue.jsx:29]: yep. will merge post #160

File: simplq/src/components/pages/CreateQueue.jsx [hunk 4]
```diff
@@ -11,76 +11,66 @@ import { SimplQHeader } from '../common/Header';
...
-  }
+const CreateQueue = ({ history }) => {
+  const [textFieldValue, setTextFieldValue] = useState('');
+  const [invalidMsg, setInvalidMsg] = useState('');
+  const [createInProgress, setCreateInProgress] = useState(false);
 
-  handleClick = (queueName) => {
-    if (this.state.textFieldValue === '') this.setState({ invalidMsg: 'Queue name is required' });
+  const handleClick = (queueName) => {
+    if (textFieldValue === '') setTextFieldValue('Queue name is required');
```
[the remark this answers is not in the document]
Comment (PavanDongare) [reply simplq/src/components/pages/CreateQueue.jsx:20]: changed

File: simplq/src/components/pages/CreateQueue.jsx (hunk 2 above)
Comment (daltonfury42) [reply simplq/src/components/pages/CreateQueue.jsx:29]: #160 is merged

File: simplq/src/components/pages/CreateQueue.jsx (hunk 1 above)
Comment (navaneeth-spotnana) [reply simplq/src/components/pages/CreateQueue.jsx:62]: Let's give. Looks like it will be string always

Comment (PavanDongare) [reply simplq/src/components/pages/CreateQueue.jsx]: changed.

**Comment by daltonfury42** [comment]
@PavanDongare feel free to merge. And congrats (and thanks) for your first PR. 🍻

**Pull request merged**